kms-infrastructure.tf

```diff
@@ -25,6 +25,12 @@ resource "aws_kms_key" "infrastructure" {
         account_id = local.infrastructure_vpc_flow_logs_s3_with_athena && local.infrastructure_kms_encryption ? local.aws_account_id : ""
         region     = local.aws_region
       }
+      )}${local.enable_infrastructure_ecs_cluster && local.infrastructure_kms_encryption ? "," : ""}
+      ${templatefile("${path.root}/policies/kms-key-policy-statements/service-sns-allow-encrypt.json.tpl",
+      {
+        sns_topic_arn = local.enable_infrastructure_ecs_cluster && local.infrastructure_kms_encryption ? "arn:aws:sns:${local.aws_region}:${local.aws_account_id}:${terraform.workspace}-infrastructure-ecs-cluster-autoscaling-lifecycle-termination" : ""
+        services      = jsonencode(["autoscaling.amazonaws.com"])
+      }
   )}
```

Comment:
This is really hard to read at a glance, do you think it's worth splitting into smaller `local`s then combining them here? Similar to what I did here https://github.com/DFE-Digital/terraform-azurerm-logic-app-workflow-slack/blob/main/locals.tf#L29-L58

Reply:
Yeah this has a weird structure because it's building a single policy file, with conditional statements, which is making it difficult to read
I felt that splitting it out into smaller locals for readability, would have the effect of making it more difficult to debug, as the conditions would be in multiple places.
The plan is, as with all the other policies, is to convert them to using the relevant policy resources rather than using template files

Reply:
OK thanks. Worth including a comment for future Chris/ops people to suggest the improvement of moving to policy resources?

Reply:
We have a card in our backlog to convert the policies 👍 
This will be included as part of that